src/Compilers/Core/Portable/Compilation.EmitStream.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    public abstract partial class Compilation
+    {
+        /// <summary>
+        /// Describes the kind of real signing that is being done during Emit. In the case of public signing
+        /// this value will be <see cref="None"/>.
+        /// </summary>
+        internal enum EmitStreamSignKind
+        {
+            None,
+            SignedWithBulider,
```

Comment:
>SignedWithBulider [](start = 12, length = 17)

It would be good to add comments describing the difference and perhaps giving examples of scenarios